tests/Unit/Evolution/Systems/NewtonianEuler/Test_Fluxes.cpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/NewtonianEuler/Fluxes.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "tests/Unit/DataStructures/TestHelpers.hpp"
+#include "tests/Unit/Pypp/Pypp.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Unit/TestingFramework.hpp"
+
+namespace {
+
+template <typename T, size_t Dim>
+void test_fluxes(const T& used_for_size) {
+  std::random_device r;
+  const auto seed = r();
+  std::mt19937 generator(seed);
+  INFO("seed = " << seed);
+  std::uniform_real_distribution<> distribution(-1.0, 1.0);
+
+  const auto nn_generator = make_not_null(&generator);
+  const auto nn_distribution = make_not_null(&distribution);
+
+  const auto momentum_density = make_with_random_values<tnsr::I<T, Dim>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto energy_density = make_with_random_values<Scalar<T>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto velocity = make_with_random_values<tnsr::I<T, Dim>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto pressure = make_with_random_values<Scalar<T>>(
+      nn_generator, nn_distribution, used_for_size);
+
+  auto mass_density_flux =
+      make_with_value<tnsr::I<DataVector, Dim>>(used_for_size, 0.0);
+  auto momentum_density_flux =
+      make_with_value<tnsr::IJ<DataVector, Dim>>(used_for_size, 0.0);
+  auto energy_density_flux =
+      make_with_value<tnsr::I<DataVector, Dim>>(used_for_size, 0.0);
+  NewtonianEuler::fluxes(make_not_null(&mass_density_flux),
+                         make_not_null(&momentum_density_flux),
+                         make_not_null(&energy_density_flux), momentum_density,
+                         energy_density, velocity, pressure);
+
+  const auto expected_mass_density_flux = pypp::call<tnsr::I<T, Dim>>(
+      "Fluxes", "mass_density_flux", momentum_density);
+  const auto expected_momentum_density_flux = pypp::call<tnsr::IJ<T, Dim>>(
+      "Fluxes", "momentum_density_flux", momentum_density, velocity, pressure);
+  const auto expected_energy_density_flux = pypp::call<tnsr::I<T, Dim>>(
+      "Fluxes", "energy_density_flux", energy_density, velocity, pressure);
+
+  for (size_t i = 0; i < Dim; ++i) {
+    CHECK_ITERABLE_APPROX(expected_mass_density_flux.get(i),
```

Comment:
You can just call CHECK_ITERABLE_APPROX on the tensors directly without the for loops since it works recursively.